app/models/comment.rb

```diff
@@ -248,8 +248,19 @@ def delete_for_user(user, reason = nil)
     self.user.refresh_counts!
   end
 
-  def deliver_mention_notifications
+  def deliver_notifications
+    notified = deliver_reply_notifications
+    deliver_mention_notifications(notified)
+  end
+
+  def deliver_mention_notifications(notified = nil)
+    notified ||= []
+
     self.plaintext_comment.scan(/\B\@([\w\-]+)/).flatten.uniq.each do |mention|
```

Comment:
Name this array like `to_notify = self.plaint....` and then update the loop to `(to_notify - notified).each do`, then you don't need the imperative if/next.